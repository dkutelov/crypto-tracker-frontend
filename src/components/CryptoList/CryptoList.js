import { useContext } from 'react';
import CryptoItem from '../CryptoItem/CryptoItem';
import styles from './CryptoList.module.css';
import CryptoContext from '../../context/cryptoContext';

const headings = [
  'Icon',
  'Crypto',
  'Symbol',
  'Price',
  'High/ Low 24h',
  'Change',
  'Volume',
  'More',
];

const CryptoList = ({ searchTerm }) => {
  let cryptoData = useContext(CryptoContext);

  if (searchTerm) {
    cryptoData = cryptoData.filter((coin) =>
      coin.id.startsWith(searchTerm.toLowerCase())
    );
  }
  return (
    <div className={styles.container}>
      <table className={styles.wrapper}>
        <thead>
          <tr>
            {headings.map((heading) => (
              <th key={heading}>{heading}</th>
            ))}
          </tr>
        </thead>
        <tbody>
          {cryptoData.length > 0 &&
            cryptoData.map((crypto) => (
              <CryptoItem key={crypto.id} {...crypto} />
            ))}
        </tbody>
      </table>
      {cryptoData.length === 0 && (
        <div style={{ textAlign: 'center', marginTop: '2em' }}>No data</div>
      )}
    </div>
  );
};

export default CryptoList;
