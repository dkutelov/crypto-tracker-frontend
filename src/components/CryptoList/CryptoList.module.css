.container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.wrapper {
  max-width: 100%;
  margin-top: 1em;
  border-spacing: 1px;
}

@media only screen and (min-width: 768px) {
  .wrapper {
    max-width: 950px;
    margin-top: 2em;
    border-spacing: 15px;
  }
}

.tableHeading {
  font-size: 0.7em;
  padding: 0.25em;
}

@media only screen and (min-width: 768px) {
  .tableHeading {
    font-size: 1em;
    padding: 0.5em 1em;
  }
}

@media only screen and (max-width: 768px) {
  .noShow {
    display: none;
  }
}
