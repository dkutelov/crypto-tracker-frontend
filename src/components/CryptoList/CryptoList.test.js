import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import CryptoList from './CryptoList';
import CryptoContext from '../../context/cryptoContext';

describe('Crypto List Component', () => {
  const items = [
    {
      id: 1,
      name: 'bitcoin',
      image: 'http://www.test.com/some-image.jpg',
      symbol: 'bitcoin',
      current_price: 123,
      high_24h: 125,
      low_24h: 121,
      price_change_percentage_24h: 5.25,
      total_volume: 123123,
    },
  ];

  it('Coin image should have correct src', () => {
    render(
      <CryptoContext.Provider value={items}>
        <BrowserRouter>
          <CryptoList searchTerm='' />
        </BrowserRouter>
      </CryptoContext.Provider>
    );

    expect(document.querySelector('img').src).toBe(
      'http://www.test.com/some-image.jpg'
    );
  });

  it('Should match snapshot', () => {
    const component = renderer.create(
      <CryptoContext.Provider value={items}>
        <BrowserRouter>
          <CryptoList searchTerm='' />
        </BrowserRouter>
      </CryptoContext.Provider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
