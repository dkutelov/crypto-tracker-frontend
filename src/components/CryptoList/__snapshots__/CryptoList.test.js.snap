// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crypto List Component Should match snapshot 1`] = `
<div
  className="container"
>
  <table
    className="wrapper"
  >
    <thead>
      <tr>
        <th>
          Icon
        </th>
        <th>
          Crypto
        </th>
        <th>
          Symbol
        </th>
        <th>
          Price
        </th>
        <th>
          High/ Low 24h
        </th>
        <th>
          Change
        </th>
        <th>
          Volume
        </th>
        <th>
          More
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="coinImageWrapper"
        >
          <img
            alt="bitcoin"
            className="coinImage"
            src="http://www.test.com/some-image.jpg"
          />
        </td>
        <td>
          bitcoin
        </td>
        <td>
          BITCOIN
        </td>
        <td
          className="coinPrice"
        >
          123
        </td>
        <td>
          125
           / 
          121
        </td>
        <td
          className="priceUp"
        >
          5.25
           %
        </td>
        <td
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          123123
        </td>
        <td>
          <a
            href="/coins/1"
            onClick={[Function]}
          >
            <img
              alt="bitcoin"
              src="more.svg"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
